### ISSUE TITLE
Assembly Resolution Issue - Compilation failures! The type or namespace name 'Results' does not exist in the namespace 'Microsoft.AspNetCore.Http'

### ISSUE THREAD

**Issue by JoshuaNitschke** [JasperFx/wolverine#549]
**STEPS TO REPRODUCE**
1) Pull repo branch "problem-details" from: https://github.com/JoshuaNitschke/wolverine-issue-reproduction/tree/problem-details
2) Run tests - one failure 
3) Run `dotnet run -- codegen write`
4) Run tests again, all pass

**EXPECTED RESULT**
Unit tests pass without having to run `dotnet run -- codegen write`

**VIDEO OF THE ABOVE STEPS**
[ProblemDetails.webm](https://github.com/JasperFx/wolverine/assets/33307206/f8c23a60-41bd-480e-975b-017ecc2b869e)

Here is the full output of the failing test:
```
Alba.ScenarioAssertionException: Expected status code 400, but was 500

Alba.ScenarioAssertionException
Expected status code 400, but was 500


Actual body text was:

System.InvalidOperationException: Compilation failures!

CS0234: The type or namespace name 'Results' does not exist in the namespace 'Microsoft.AspNetCore.Http' (are you missing an assembly reference?)

Code:

// <auto-generated/>
#pragma warning disable
using Microsoft.AspNetCore.Routing;
using System;
using System.Linq;
using Wolverine.Http;

namespace Internal.Generated.WolverineHandlers
{
    // START: GET_problem_details_1
    public class GET_problem_details_1 : Wolverine.Http.HttpHandler
    {
        private readonly Wolverine.Http.WolverineHttpOptions _wolverineHttpOptions;

        public GET_problem_details_1(Wolverine.Http.WolverineHttpOptions wolverineHttpOptions) : base(wolverineHttpOptions)
        {
            _wolverineHttpOptions = wolverineHttpOptions;
        }



        public override async System.Threading.Tasks.Task Handle(Microsoft.AspNetCore.Http.HttpContext httpContext)
        {
            var problemDetails1Endpoint = new WolverineIssueReproduction.WebApi.Controllers.ProblemDetails1Endpoint();
            var problemDetails = problemDetails1Endpoint.Before();
            // Evaluate whether the processing should stop if there are any problems
            if (!(ReferenceEquals(problemDetails, Wolverine.Http.WolverineContinue.NoProblems)))
            {
                await Microsoft.AspNetCore.Http.Results.Problem(problemDetails).ExecuteAsync(httpContext).ConfigureAwait(false);
                return;
            }


            var result_of_Get = problemDetails1Endpoint.Get();
            await WriteString(httpContext, result_of_Get);
        }

    }

    // END: GET_problem_details_1
    
    
}


   at JasperFx.RuntimeCompiler.AssemblyGenerator.Generate(String code)
   at JasperFx.RuntimeCompiler.AssemblyGenerator.Compile(GeneratedAssembly generatedAssembly, IServiceVariableSource services)
   at JasperFx.RuntimeCompiler.CodeFileExtensions.InitializeSynchronously(ICodeFile file, GenerationRules rules, ICodeFileCollection parent, IServiceProvider services)
   at Wolverine.Http.HttpChain.<>c__DisplayClass67_0.<BuildEndpoint>b__0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at Wolverine.Http.HttpChain.<>c__DisplayClass67_0.<BuildEndpoint>b__4(HttpContext c)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.Invoke(HttpContext httpContext)
--- End of stack trace from previous location ---
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)

HEADERS
=======
Host: localhost


   at Alba.ScenarioAssertionException.AssertAll() in /_/src/Alba/ScenarioAssertionException.cs:line 35
   at Alba.Scenario.RunAssertions(HttpContext context) in /_/src/Alba/Scenario.cs:line 280
   at Alba.AlbaHost.Scenario(Action`1 configure) in /_/src/Alba/AlbaHost.cs:line 251
   at Alba.AlbaHost.Scenario(Action`1 configure) in /_/src/Alba/AlbaHost.cs:line 255
   at WolverineIssueReproduction.Tests.AppFixture.<>c__DisplayClass8_0.<<TrackedHttpCall>b__0>d.MoveNext() in D:\GIT\wolverine-issue-reproduction\src\WolverineIssueReproduction.Tests\AppFixture.cs:line 62
--- End of stack trace from previous location ---
   at Wolverine.Tracking.TrackedSession.ExecuteAndTrackAsync()
   at Wolverine.Tracking.TrackedSession.ExecuteAndTrackAsync()
   at Wolverine.Tracking.WolverineHostMessageTrackingExtensions.ExecuteAndWaitAsync(IHost host, Func`2 action, Int32 timeoutInMilliseconds)
   at WolverineIssueReproduction.Tests.AppFixture.TrackedHttpCall(Action`1 configuration) in D:\GIT\wolverine-issue-reproduction\src\WolverineIssueReproduction.Tests\AppFixture.cs:line 58
   at WolverineIssueReproduction.Tests.AppFixture.should_not_have_results_does_not_exist_in_namespace_error() in D:\GIT\wolverine-issue-reproduction\src\WolverineIssueReproduction.Tests\AppFixture.cs:line 29
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_0.<<InvokeTestMethodAsync>b__1>d.MoveNext() in /_/src/xunit.execution/Sdk/Frameworks/Runners/TestInvoker.cs:line 264
--- End of stack trace from previous location ---
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in /_/src/xunit.execution/Sdk/Frameworks/ExecutionTimer.cs:line 48
   at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in /_/src/xunit.core/Sdk/ExceptionAggregator.cs:line 90
```

**Comment by jeremydmiller** [comment]
@JoshuaNitschke *I* hit this on something new while changing a code sample today, so I've got a reproduction just fine now

**Comment by jeremydmiller** [comment]
Easy money. Fix coming in 1.7 this week/early next week.

**Comment by JoshuaNitschke** [comment]
Excellent, thanks @jeremydmiller
